src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java

```diff
@@ -460,7 +563,7 @@ private void waitForRecovery(ShutdownNotifier notifier) {
     }
 
     private boolean isConnectionRecovery(ShutdownNotifier notifier) {
-        return !notifier.isOpen() && !connectionIsClosed.get();
+        return !(notifier instanceof AutorecoveringChannel) && !notifier.isOpen() && !connectionIsClosed.get();
```

Comment:
Why have we added this check here?

Reply:
I assume that we should wait for reconnection only for autorecovering connections. But looks like the condition is wrong. Probably correct condition is `(notifier instanceof AutorecoveringConnection) && !notifier.isOpen() && !connectionIsClosed.get()`.